### PR TITLE
TDR-2112 Add files status mutation to add a FileStatusRow for a file

### PR DESCRIPTION
**Pull request by vimleshtna**

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/GraphQlTypes.scala [hunk 1]
```diff
@@ -20,7 +20,8 @@ object GraphQlTypes {
       AntivirusMetadataFields.mutationFields ++
       FileMetadataFields.mutationFields ++
       FFIDMetadataFields.mutationFields ++
-      FinalTransferConfirmationFields.mutationFields
+      FinalTransferConfirmationFields.mutationFields ++
+      fields.FileStatusFields.mutationFields
```
Comment (MancunianSam) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/GraphQlTypes.scala:24]: I think you can remove the `fields` prefix here as the whole object has been imported.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
+  }
+
+  private def validateStatusTypeAndValue(addFileStatusInput: AddFileStatusInput) = {
+    val statusType: String = addFileStatusInput.statusType
+    val statusValue: String = addFileStatusInput.statusValue
+
+    if (Upload == statusType && List(Success, Failed).contains(statusValue)) {
+      true
+    } else {
+      throw InputDataException(s"Invalid FileStatus input: either '$statusType' or '$statusValue'")
```
Comment (MancunianSam) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala:35]: This is personal preference so feel free to ignore me. Throwing exceptions isn't very functional (I know there's a lot thrown in here already)
If you return `Future[Boolean]` from this method then you can do 
```scala
    if (Upload == statusType && List(Success, Failed).contains(statusValue)) {
      Future(true)
    } else {
      Future.failed(InputDataException(s"Invalid FileStatus input: either '$statusType' or '$statusValue'"))
    }
```

That will let you use it inside your for comprehension as well
```scala
    for {
      _ <- validateStatusTypeAndValue(addFileStatusInput)
      _ <- fileStatusRepository.addFileStatuses(List(FilestatusRow(uuidSource.uuid,...
```

File: src/test/resources/json/addFileStatus_data_all.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "data": {
+    "addFileStatus": {
+      "fileId": "07a3a4bd-0281-4a6d-a4c1-8fa3239e1313",
+      "statusType": "Upload",
+      "statusValue": "Success"
+    }
+  }
+}
```
Comment (MancunianSam) [inline src/test/resources/json/addFileStatus_data_all.json:9]: Can you add a line break at the end of the file please.

File: src/test/resources/json/addFileStatus_mutation_alldata.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "query": "mutation addFileStatus($addFileStatusInput: AddFileStatusInput!){addFileStatus(addFileStatusInput: $addFileStatusInput){fileId,statusType,statusValue}}",
+  "variables": {
+    "addFileStatusInput": {
+      "fileId": "07a3a4bd-0281-4a6d-a4c1-8fa3239e1313",
+      "statusType": "Upload",
+      "statusValue": "Success"
+    }
+  }
+}
```
Comment (MancunianSam) [inline src/test/resources/json/addFileStatus_mutation_alldata.json:10]: Another line break here please. There's a few missing from these json files, if you could add them all in.

**Review by MancunianSam — COMMENTED** [review]
Looks good, thanks. Just a couple of small things.

File: src/test/resources/json/addFileStatus_mutation_alldata.json (hunk 4 above)
Comment (vimleshtna) [reply src/test/resources/json/addFileStatus_mutation_alldata.json]: Yeah, added in all the jsons, thanks.

File: src/test/resources/json/addFileStatus_data_all.json (hunk 3 above)
Comment (vimleshtna) [reply src/test/resources/json/addFileStatus_data_all.json]: Added, thanks.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala (hunk 2 above)
Comment (vimleshtna) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala]: Yeah, make sense. Thanks.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/GraphQlTypes.scala (hunk 1 above)
Comment (vimleshtna) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/GraphQlTypes.scala]: Right, I have removed the `fields` prefix, thanks

**Review by MancunianSam — APPROVED** [review]

**Pull request merged**